lisa-kernel/src/main/scala/lisa/kernel/fol/CommonDefinitions.scala

```diff
@@ -11,39 +11,47 @@ private[fol] trait CommonDefinitions {
    */
   protected trait Label {
     val arity: Int
-    val id: String
+    val id: Identifier
   }
 
-  /**
-   * \begin{lstlisting}[language=Scala, frame=single]
-   * object MyTheoryName extends lisa.Main {
-   *
-   *  THEOREM("theoremName") of "desired conclusion" PROOF {
-   *
-   *    ... : Proof
-   *
-   *  } using (listOfJustifications)
-   *  show
-   * }
-   * \end{lstlisting}
-   * A constant label can represent a fixed symbol of a theory or a logical symbol
-   */
-  trait ConstantLabel extends Label
+  sealed case class Identifier(val name: String, val no: Int) {
+    require(no >> 31 == 0, "Variable index must be positive")
```

Comment:
This is not appreciably faster than a simple comparison, and unnecessarily assumes a 32-bit Int. This is the case in the Java spec but reduces readability and may cause problems if we ever compile outside the JVM.

```scala
scala> time {for (x <- -9999999 to 9999999) throwaway(x >> 31 == 0)}
102463 microseconds

scala> time {for (x <- -9999999 to 9999999) throwaway(x >> 31 == 0)}
114483 microseconds

scala> time {for (x <- -9999999 to 9999999) throwaway(x > 0)}
104077 microseconds

scala> time {for (x <- -9999999 to 9999999) throwaway(x > 0)}
109662 microseconds
```

Here `throwaway(x) = ()`.

Reply:
Noted